tests/test_config.py

```diff
@@ -5,18 +5,42 @@
 import click
 import pytest
 
+from snakefmt import DEFAULT_LINE_LENGTH
 from snakefmt.exceptions import MalformattedToml
 from snakefmt.formatter import TAB
-from snakefmt.snakefmt import main, read_snakefmt_defaults_from_pyproject_toml
+from snakefmt.snakefmt import (
+    find_pyproject_toml,
+    inject_snakefmt_config,
+    main,
+    read_snakefmt_config,
+)
 from tests import setup_formatter
 
 
+def test_black_and_snakefmt_default_line_lengths_aligned():
+    assert DEFAULT_LINE_LENGTH == black.DEFAULT_LINE_LENGTH
+
+
+class TestFindPyprojectToml:
+    def test_find_pyproject_toml_nested_directory(self, tmp_path):
+        config_file = (tmp_path / "pyproject.toml").resolve()
+        config_file.touch()
+        dir1 = Path(tmp_path / "dir1").resolve()
+        dir1.mkdir()
+        snakefile = dir1 / "Snakefile"
+        actual = find_pyproject_toml((str(snakefile),))
+        assert actual == str(config_file)
+
+
 class TestConfigAdherence:
+    def test_no_config_path_empty_config_dict(self):
+        parsed_config = read_snakefmt_config(None)
+        assert parsed_config == dict()
+
     def test_config_adherence_for_python_outside_rules(self, cli_runner, tmp_path):
         stdin = "include: 'a'\nlist_of_lots_of_things = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]"
-        line_length = 30
         config = tmp_path / "pyproject.toml"
-        config.write_text(f"[tool.snakefmt]\nline_length = {line_length}\n")
+        config.write_text("[tool.snakefmt]\nline_length = 30\n")
```

Comment:
This makes the test harder to read/adjust I think

Reply:
Agreed and changed